tests/php/View/SSViewerTest.php

```diff
@@ -1331,6 +1374,88 @@ public function testCastingHelpers()
         );
     }
 
+    public function provideLoop(): array
+    {
+        return [
+            'nested array and iterator' => [
+                'iterable' => [
+                    [
+                        'value 1',
+                        'value 2',
+                    ],
+                    new ArrayIterator([
+                        'value 3',
+                        'value 4',
+                    ]),
```

Comment:
I'm surprised this works - but I think it's worth keeping it working, since it doesn't require much additional complexity to maintain.